runtime/src/main/java/org/corfudb/runtime/view/SnowflakeGuidFactory.java

```diff
@@ -0,0 +1,49 @@
+package org.corfudb.runtime.view;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Fast Globally Unique Identity (GUID) generator that returns compact ids
+ * having a notion of a universal comparable ordering.
+ *
+ *    ______________________________________________
+ *   | timestamp lower 42 bits | Node Id | Sequence |
+ *    ------------------------- --------- ----------
+ *    <------ 42 bits ---------><-10bits-><-12bits-->
+ *
```

Comment:
What's the difference between a timestamp and a sequence?

Reply:
Just the resolution. Timestamp is coarse grained and the sequence helps distinguish several consecutive ids within that same time quantum.